hbase-agent/src/main/java/org/apache/ranger/authorization/hbase/RangerAuthorizationCoprocessor.java

```diff
@@ -491,14 +515,52 @@ ColumnFamilyAccessResult evaluateAccess(ObserverContext<?> ctx, String operation
 				session.resourceMatchingScope(RangerAccessRequest.ResourceMatchingScope.SELF);
 				session.ignoreDescendantDeny(true);
 			} else {
-				LOG.debug("evaluateAccess: columns collection not empty.  Skipping Family level check, will do finer level access check.");
+				boolean isColumnAuthShortCircuitingEnabled = hbasePlugin.getPropertyIsColumnAuthOptimizationEnabled();
+				if(LOG.isDebugEnabled()) {
+					LOG.debug("evaluateAccess: columns collection not empty." +
+							" Skipping Family level check, will do finer level access check for columns.");
+				}
+				if (isColumnAuthShortCircuitingEnabled) {
+					session.column(null)
+							.buildRequest()
+							.authorize();
+
+					boolean isColumnFamilyAuthorized = session.isAuthorized();
+
+					//check if column family fully authorized i.e. no deny for columns
+					session.column(null)
```

Comment:
Can the second authz, starting with line 531, be skipped when `isColumnFamilyAuthorized` is `false`?

```
boolean isColumnFamilyAndDescendantsAuthorized = false;

if (session.isAuthorized()) {
  session.column(null)
         .resourceMatchingScope(RangerAccessRequest.ResourceMatchingScope.SELF_OR_DESCENDANTS)
         .ignoreDescendantDeny(false)
         .buildRequest()
         .authorize();

  isColumnFamilyAndDescendantsAuthorized = session.isAuthorized();
}